コンパイル時warningがでる

```
/home/ubuntu/src/php-src/ext/study_extension/study_extension.c: In function ‘study_extension_var_dump’:
/home/ubuntu/src/php-src/ext/study_extension/study_extension.c:73:53: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘zend_long’ {aka ‘long int’} [-Wformat=]
   73 |                         php_printf("%sRESOURCE: id=%d type=%s\n", STUDY_COMMON, Z_RES_P(struc)->handle, type_name ? type_name : "Unknown");
      |                                                    ~^
      |                                                     |
      |                                                     int
      |                                                    %ld
/home/ubuntu/src/php-src/ext/study_extension/study_extension.c: In function ‘study_php_print_info’:
/home/ubuntu/src/php-src/ext/study_extension/study_extension.c:538:38: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  538 |                 char *zend_version = get_zend_version();
      |                                      ^~~~~~~~~~~~~~~~
/home/ubuntu/src/php-src/ext/study_extension/study_extension.c:741:50: warning: passing argument 2 of ‘zend_hash_sort’ from incompatible pointer type [-Wincompatible-pointer-types]
  741 |                 zend_hash_sort(&sorted_registry, module_name_cmp, 0);
      |                                                  ^~~~~~~~~~~~~~~
      |                                                  |
      |                                                  int (*)(const void *, const void *)
In file included from /home/ubuntu/src/php-src/Zend/zend.h:33,
                 from /home/ubuntu/src/php-src/main/php.h:31,
                 from /home/ubuntu/src/php-src/ext/study_extension/study_extension.c:9:
/home/ubuntu/src/php-src/Zend/zend_hash.h:301:98: note: expected ‘bucket_compare_func_t’ {aka ‘int (*)(struct _Bucket *, struct _Bucket *)’} but argument is of type ‘int (*)(const void *, const void *)’
  301 | static zend_always_inline void ZEND_FASTCALL zend_hash_sort(HashTable *ht, bucket_compare_func_t compare_func, bool renumber) {
      |                                                                            ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
```

--- a/study_extension.c
+++ b/study_extension.c
@@ -70,7 +70,7 @@ again:
 			break;
 		case IS_RESOURCE: {
 			const char *type_name = zend_rsrc_list_get_rsrc_type(Z_RES_P(struc));
-			php_printf("%sRESOURCE: id=%d type=%s\n", STUDY_COMMON, Z_RES_P(struc)->handle, type_name ? type_name : "Unknown");
+			php_printf("%sRESOURCE: id=%ld type=%s\n", STUDY_COMMON, Z_RES_P(struc)->handle, type_name ? type_name : "Unknown");
 			break;
 		}
 		case IS_ARRAY:
@@ -535,7 +535,7 @@ PHPAPI ZEND_COLD void study_php_print_info(int flag)
 	}
 
 	if (flag & PHP_INFO_GENERAL) {
-		char *zend_version = get_zend_version();
+		const char *zend_version = get_zend_version();
 		char temp_api[10];
 
 		php_uname = php_get_uname('a');
@@ -738,7 +738,7 @@ PHPAPI ZEND_COLD void study_php_print_info(int flag)
 
 		zend_hash_init(&sorted_registry, zend_hash_num_elements(&module_registry), NULL, NULL, 1);
 		zend_hash_copy(&sorted_registry, &module_registry, NULL);
-		zend_hash_sort(&sorted_registry, module_name_cmp, 0);
+		zend_hash_sort(&sorted_registry, (bucket_compare_func_t) module_name_cmp, 0);
 
 		ZEND_HASH_FOREACH_PTR(&sorted_registry, module) {
 			if (module->info_func || module->version) {

